encryption/src/main/kotlin/io/provenance/scope/encryption/crypto/SignerFactory.kt

```diff
@@ -8,14 +8,14 @@ import java.security.KeyPair
 import java.security.PublicKey
 
 class SignerFactory(
-//    private val signAndVerifyApi: SignAndVerifyApi? = null
+    private val signAndVerifyApi: SignAndVerifyApi? = null
```

Comment:
Rather than having this be nullable here it can be moved to the `SmartKeyRef` class.

Reply:
Oh, I guess that would solve the question of having a SignerFactory vs the extension function like you have here https://github.com/provenance-io/p8e-scope-sdk/pull/18/files#diff-8dc189609cd7f54b607a835cf2970a969f715f442ea71e731e01bca5dbf8386eR15 ... and also allows there to be multiple SmartKey api instances used per key if really desired...

Reply:
Yeah, that's a nice benefit as well.

Reply:
Should I pull in the branch listed here?

Reply:
I think maybe we should get that other branch merged into main first and then just update this from there, and resolve the inevitable conflicts